server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -433,33 +441,35 @@ public void poll()
 
       log.debug("Starting polling of segment table");
 
-      final List<DataSegment> segments = dbi.withHandle(
-          new HandleCallback<List<DataSegment>>()
+      // some databases such as PostgreSQL require auto-commit turned off
+      // to stream results back, enabling transactions disables auto-commit
+      final List<DataSegment> segments = connector.getDBI().inTransaction(
```

Comment:
once we set the connection to read-only it will be effectively the same as a single statement with auto-commit set to true